src/main/scala/edu/ie3/osmogrid/guardian/OsmoGridGuardian.scala

```diff
@@ -7,131 +7,89 @@
 
 import akka.actor.typed.Behavior
-import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
 import akka.actor.typed.ActorRef
 import edu.ie3.datamodel.models.input.container.{
   JointGridContainer,
   SubGridContainer
 }
 import edu.ie3.datamodel.utils.ContainerUtils
-import edu.ie3.osmogrid.cfg.OsmoGridConfig
-import edu.ie3.osmogrid.cfg.OsmoGridConfig.Generation
+import edu.ie3.osmogrid.cfg.{ConfigFailFast, OsmoGridConfig}
+import edu.ie3.osmogrid.cfg.OsmoGridConfig.{Generation, Output}
 import edu.ie3.osmogrid.io.input.InputDataProvider
-import edu.ie3.osmogrid.io.input.InputDataProvider.InputDataEvent
 import edu.ie3.osmogrid.io.output.ResultListener
-import edu.ie3.osmogrid.io.output.ResultListener.{GridResult, ResultEvent}
+import edu.ie3.osmogrid.io.output.ResultListener.{GridResult, Request}
 import edu.ie3.osmogrid.lv.LvCoordinator
 import edu.ie3.osmogrid.lv.LvCoordinator.ReqLvGrids
+import org.slf4j.Logger
 
+import java.util.UUID
 import scala.jdk.CollectionConverters.*
 import scala.util.{Failure, Success, Try}
 
 object OsmoGridGuardian {
 
-  sealed trait OsmoGridGuardianEvent
-  final case class Run(cfg: OsmoGridConfig) extends OsmoGridGuardianEvent
-  object InputDataProviderDied extends OsmoGridGuardianEvent
-  object ResultEventListenerDied extends OsmoGridGuardianEvent
-  object LvCoordinatorDied extends OsmoGridGuardianEvent
-  final case class RepLvGrids(grids: Vector[SubGridContainer])
-      extends OsmoGridGuardianEvent
+  /* Messages, that are understood and sent */
+  sealed trait Request
 
-  def apply(): Behavior[OsmoGridGuardianEvent] = idle()
+  /** Message to initiate a grid generation run
+    *
+    * @param cfg
+    *   Configuration for the tool
+    * @param additionalListener
+    *   Addresses of additional listeners to be informed about results
+    * @param runId
+    *   Unique identifier for that generation run
+    */
+  final case class Run(
+      cfg: OsmoGridConfig,
+      additionalListener: Seq[ActorRef[ResultListener.ResultEvent]] = Seq.empty,
+      runId: UUID = UUID.randomUUID()
+  ) extends Request
 
-  private def idle(): Behavior[OsmoGridGuardianEvent] =
-    Behaviors.receive { (ctx, msg) =>
-      msg match {
-        case Run(cfg) =>
-          ctx.log.info("Initializing grid generation!")
+  /* dead watch events */
+  sealed trait Watch extends Request {
+    val runId: UUID
+  }
 
-          ctx.log.info("Starting input data provider")
-          val inputProvider =
-            ctx.spawn(InputDataProvider(cfg.input), "InputDataProvider")
-          ctx.watchWith(inputProvider, InputDataProviderDied)
-          ctx.log.debug("Starting output data listener")
-          val resultEventListener =
-            ctx.spawn(ResultListener(cfg.output), "ResultListener")
-          ctx.watchWith(resultEventListener, ResultEventListenerDied)
+  private[guardian] final case class RunGuardianDied(override val runId: UUID)
```

Comment:
Why package private? Not required, can be just private

```suggestion
  private final case class RunGuardianDied(override val runId: UUID)
```